### PR TITLE
Optimize configuration and layout to reduce build-time

### PR DESCRIPTION
**Pull request by thaJeztah**
Various changes in configuration and some of the templates to considerably reduce build time:

- enable the `--profile` option to output more detailed information about build-times
- inline side-menu include.
  It was fully static, only used in a single place, but would be processed for each page.
- Optimize configuration to reduce build-times
    - remove wildcard match for defaults (this caused a ripple effect and slowed down total build time with 60 seconds?)
    - skip generating metadata.json (which took 25+ seconds)
    - skip generating sitemap.xml (10+ seconds)
- inline "body-landing" and "body" into layout
  This reduces the time to build, as Jekyll doesn't have to render the body, then copy it into the layout.

With the changes in this PR combined :

Development before: 121 seconds
Development after:   33 seconds
Production before:  132 seconds
Production after:    46 seconds

Detailed info:

DEVELOPMENT (before)
-----------------------------

Filename                                                 | Count |     Bytes |   Time
---------------------------------------------------------|-------|-----------|-------
js/metadata.json                                         |     1 |   125.49K | 24.837
sitemap.xml                                              |     1 |    36.36K | 12.301
_layouts/docs.html                                       |   688 | 18957.10K |  7.982
_includes/body.html                                      |   687 | 16370.57K |  6.041
_includes/toc_pure_liquid.html                           |   646 |   527.40K |  3.852
_includes/head.html                                      |   688 |  2497.12K |  1.517
_includes/cli.md                                         |   234 |  1318.62K |  1.132
_includes/header.html                                    |   687 |  1448.47K |  0.297
_layouts/redirect.html                                   |   691 |   319.10K |  0.292
_includes/read_time.html                                 |   431 |    21.15K |  0.271
_includes/navigation.html                                |   687 |  1153.27K |  0.211
_includes/footer.html                                    |   688 |  3201.48K |  0.064
glossary.md                                              |     1 |    23.82K |  0.045
_includes/side-menu.html                                 |   687 |   101.31K |  0.021
get-started/orchestration.md                             |     1 |    18.05K |  0.019
_includes/theme-switch.html                              |   688 |   255.31K |  0.018
engine/reference/commandline/container_run.md            |     1 |    25.31K |  0.018
engine/reference/commandline/run.md                      |     1 |    49.21K |  0.018
engine/reference/commandline/create.md                   |     1 |    25.12K |  0.017
engine/reference/commandline/container_create.md         |     1 |    24.74K |  0.017
engine/reference/commandline/service_update.md           |     1 |    34.10K |  0.016
engine/reference/commandline/service_create.md           |     1 |    60.19K |  0.016
engine/reference/commandline/builder_build.md            |     1 |    10.05K |  0.014
engine/reference/commandline/build.md                    |     1 |    36.78K |  0.010
engine/reference/commandline/container_stats.md          |     1 |     6.39K |  0.010
engine/reference/commandline/image_build.md              |     1 |    11.35K |  0.009
engine/reference/commandline/swarm_init.md               |     1 |    13.08K |  0.009
engine/reference/commandline/buildx_build.md             |     1 |    10.24K |  0.008
engine/reference/commandline/swarm.md                    |     1 |     2.69K |  0.008
engine/reference/commandline/container_update.md         |     1 |     9.17K |  0.008
engine/reference/commandline/plugin_disable.md           |     1 |     4.28K |  0.007
engine/reference/commandline/app_bundle.md               |     1 |     4.68K |  0.007
engine/reference/commandline/buildx_imagetools_create.md |     1 |     2.66K |  0.007
engine/reference/commandline/image_history.md            |     1 |     3.72K |  0.007
engine/reference/commandline/manifest.md                 |     1 |    12.03K |  0.007
engine/reference/commandline/network_prune.md            |     1 |     5.83K |  0.007
engine/reference/commandline/docker.md                   |     1 |     9.75K |  0.007
engine/reference/commandline/network_create.md           |     1 |    14.63K |  0.007
engine/reference/commandline/container_prune.md          |     1 |    10.42K |  0.007
engine/reference/commandline/volume_rm.md                |     1 |     2.83K |  0.007
engine/reference/commandline/container_start.md          |     1 |     7.09K |  0.006
engine/reference/commandline/container_logs.md           |     1 |     6.96K |  0.006
engine/reference/commandline/container_kill.md           |     1 |     5.84K |  0.006
engine/reference/commandline/stack_ps.md                 |     1 |    14.78K |  0.006
engine/reference/commandline/container_rename.md         |     1 |     5.45K |  0.006
engine/reference/commandline/exec.md                     |     1 |     5.12K |  0.006
engine/reference/commandline/container_cp.md             |     1 |     6.39K |  0.006
engine/reference/commandline/container_unpause.md        |     1 |     5.49K |  0.006
engine/reference/commandline/container_ls.md             |     1 |     7.04K |  0.006
engine/reference/commandline/container_wait.md           |     1 |     5.50K |  0.006

                    done in 121.123 seconds.

PRODUCTION (before)
-----------------------------

Filename                                            | Count |     Bytes |   Time
----------------------------------------------------|-------|-----------|-------
js/metadata.json                                    |     1 |   125.49K | 22.582
sitemap.xml                                         |     1 |    36.46K | 11.521
_layouts/docs.html                                  |   788 | 21087.70K |  8.111
_includes/toc_pure_liquid.html                      |   747 |   527.59K |  3.716
_includes/head.html                                 |   789 |  3165.82K |  1.766
_includes/cli.md                                    |   234 |  1318.62K |  1.042
_layouts/redirect.html                              |  1138 |   534.03K |  0.355
_includes/header.html                               |   788 |  1661.42K |  0.312
_includes/analytics/polldaddy.html                  |   788 |   547.60K |  0.229
_includes/navigation.html                           |   788 |  1322.82K |  0.223
_includes/analytics/google_analytics.html           |   789 |   305.89K |  0.114
_includes/footer.html                               |   789 |  3671.47K |  0.060
glossary.md                                         |     1 |    23.82K |  0.039
_includes/theme-switch.html                         |   789 |   292.79K |  0.022
engine/reference/commandline/container_run.md       |     1 |    25.31K |  0.019
engine/reference/commandline/container_create.md    |     1 |    24.74K |  0.017
get-started/orchestration.md                        |     1 |    18.05K |  0.017
engine/reference/commandline/run.md                 |     1 |    49.21K |  0.017
engine/reference/commandline/service_update.md      |     1 |    34.10K |  0.016
engine/reference/commandline/service_create.md      |     1 |    60.19K |  0.015
engine/reference/commandline/create.md              |     1 |    25.12K |  0.015
engine/reference/commandline/buildx_build.md        |     1 |    10.24K |  0.009
engine/reference/commandline/build.md               |     1 |    36.78K |  0.009
engine/reference/commandline/image_build.md         |     1 |    11.35K |  0.009
engine/reference/commandline/checkpoint_create.md   |     1 |     3.62K |  0.009
engine/reference/commandline/container_diff.md      |     1 |     5.49K |  0.008
engine/reference/commandline/buildx_bake.md         |     1 |     4.92K |  0.008
engine/reference/commandline/builder_build.md       |     1 |    10.05K |  0.008
engine/reference/commandline/trust_signer_remove.md |     1 |     6.50K |  0.007
engine/reference/commandline/stack_ps.md            |     1 |    14.78K |  0.006
engine/reference/commandline/container_export.md    |     1 |     5.83K |  0.006
engine/reference/commandline/plugin_ls.md           |     1 |     6.21K |  0.006
engine/reference/commandline/node_rm.md             |     1 |     4.36K |  0.006
engine/reference/commandline/app_install.md         |     1 |     6.88K |  0.006
engine/reference/commandline/network_create.md      |     1 |    14.63K |  0.006
engine/reference/commandline/container_exec.md      |     1 |     7.48K |  0.006
engine/reference/commandline/container_inspect.md   |     1 |     6.03K |  0.006
engine/reference/commandline/swarm_init.md          |     1 |    13.08K |  0.006
engine/reference/commandline/container_ls.md        |     1 |     7.04K |  0.006
engine/reference/commandline/container_update.md    |     1 |     9.17K |  0.006
engine/reference/commandline/plugin_install.md      |     1 |     4.97K |  0.006
engine/reference/commandline/swarm_ca.md            |     1 |     7.92K |  0.006
engine/reference/commandline/trust_signer.md        |     1 |     2.07K |  0.006
engine/reference/commandline/buildx_create.md       |     1 |     5.17K |  0.005
engine/reference/commandline/node_update.md         |     1 |     4.96K |  0.005
engine/reference/commandline/container_logs.md      |     1 |     6.96K |  0.005
engine/reference/commandline/image_ls.md            |     1 |     4.04K |  0.005
engine/reference/commandline/docker.md              |     1 |     9.75K |  0.005
engine/reference/commandline/swarm_leave.md         |     1 |     4.39K |  0.005
engine/reference/commandline/swarm_join-token.md    |     1 |     6.13K |  0.005

                    done in 131.638 seconds.

DEVELOPMENT (AFTER)
------------------------

Filename                                          | Count |     Bytes |  Time
--------------------------------------------------|-------|-----------|------
_layouts/docs.html                                |   687 | 18899.53K | 7.262
_includes/toc_pure_liquid.html                    |   646 |   527.40K | 3.652
_includes/head.html                               |   688 |  2494.68K | 1.423
_includes/cli.md                                  |   234 |  1318.62K | 1.085
_includes/header.html                             |   687 |  1448.47K | 0.284
_includes/navigation.html                         |   687 |  1153.27K | 0.198
_includes/footer.html                             |   688 |  3201.48K | 0.056
glossary.md                                       |     1 |    23.82K | 0.044
_includes/theme-switch.html                       |   688 |   255.31K | 0.021
engine/reference/commandline/run.md               |     1 |    49.21K | 0.019
engine/reference/commandline/container_run.md     |     1 |    25.31K | 0.018
engine/reference/commandline/service_update.md    |     1 |    34.10K | 0.017
engine/reference/commandline/container_create.md  |     1 |    24.74K | 0.017
get-started/orchestration.md                      |     1 |    18.05K | 0.016
engine/reference/commandline/create.md            |     1 |    25.12K | 0.016
engine/reference/commandline/service_create.md    |     1 |    60.19K | 0.014
engine/reference/commandline/buildx_create.md     |     1 |     5.17K | 0.012
engine/reference/commandline/build.md             |     1 |    36.78K | 0.010
engine/reference/commandline/app_inspect.md       |     1 |     5.20K | 0.010
engine/reference/commandline/buildx_build.md      |     1 |    10.24K | 0.009
engine/reference/commandline/container_restart.md |     1 |     5.85K | 0.009
engine/reference/commandline/image_build.md       |     1 |    11.35K | 0.009
engine/reference/commandline/unpause.md           |     1 |     1.07K | 0.008
engine/reference/commandline/container_update.md  |     1 |     9.17K | 0.008
engine/reference/commandline/image.md             |     1 |     2.88K | 0.007
engine/reference/commandline/node_inspect.md      |     1 |     7.64K | 0.007
engine/reference/commandline/network_create.md    |     1 |    14.63K | 0.007
engine/reference/commandline/builder_build.md     |     1 |    10.05K | 0.007
engine/reference/commandline/container_logs.md    |     1 |     6.96K | 0.007
engine/reference/commandline/context_export.md    |     1 |     2.80K | 0.007
engine/reference/commandline/container_start.md   |     1 |     7.09K | 0.006
engine/reference/commandline/container_ls.md      |     1 |     7.04K | 0.006
engine/reference/commandline/container_diff.md    |     1 |     5.49K | 0.006
engine/reference/commandline/buildx_bake.md       |     1 |     4.92K | 0.006
engine/reference/commandline/container_inspect.md |     1 |     6.03K | 0.006
engine/reference/commandline/app_install.md       |     1 |     6.88K | 0.006
engine/reference/commandline/app_upgrade.md       |     1 |     6.05K | 0.006
engine/reference/commandline/container_rename.md  |     1 |     5.45K | 0.006
engine/reference/commandline/service_logs.md      |     1 |     7.58K | 0.006
engine/reference/commandline/docker.md            |     1 |     9.75K | 0.006
engine/reference/commandline/swarm_init.md        |     1 |    13.08K | 0.006
engine/reference/commandline/app_render.md        |     1 |     5.58K | 0.006
engine/reference/commandline/update.md            |     1 |     7.33K | 0.006
engine/reference/commandline/container_exec.md    |     1 |     7.48K | 0.006
engine/reference/commandline/trust_inspect.md     |     1 |    14.73K | 0.006
engine/reference/commandline/context_create.md    |     1 |     5.68K | 0.005
engine/reference/commandline/app_status.md        |     1 |     5.30K | 0.005
engine/reference/commandline/container_export.md  |     1 |     5.83K | 0.005
engine/reference/commandline/app_init.md          |     1 |     5.53K | 0.005
engine/reference/commandline/swarm_update.md      |     1 |     5.02K | 0.005

                    done in 32.654 seconds.

PRODUCTION (AFTER)
-------------------------

Filename                                          | Count |     Bytes |  Time
--------------------------------------------------|-------|-----------|------
_layouts/docs.html                                |   788 | 21197.88K | 8.282
_includes/toc_pure_liquid.html                    |   747 |   527.59K | 3.779
js/metadata.json                                  |     1 |   125.49K | 2.445
_includes/head.html                               |   788 |  3162.05K | 1.804
sitemap.xml                                       |     1 |    36.46K | 1.349
_includes/cli.md                                  |   234 |  1318.62K | 1.048
_includes/header.html                             |   788 |  1661.42K | 0.315
_layouts/redirect.html                            |  1138 |   534.03K | 0.314
_includes/analytics/polldaddy.html                |   788 |   547.60K | 0.241
_includes/navigation.html                         |   788 |  1322.82K | 0.226
_includes/analytics/google_analytics.html         |   788 |   305.50K | 0.118
_includes/footer.html                             |   789 |  3671.47K | 0.063
glossary.md                                       |     1 |    23.82K | 0.038
_includes/theme-switch.html                       |   789 |   292.79K | 0.022
engine/reference/commandline/run.md               |     1 |    49.21K | 0.019
engine/reference/commandline/container_run.md     |     1 |    25.31K | 0.017
engine/reference/commandline/service_update.md    |     1 |    34.10K | 0.017
get-started/orchestration.md                      |     1 |    18.05K | 0.016
engine/reference/commandline/container_create.md  |     1 |    24.74K | 0.016
engine/reference/commandline/create.md            |     1 |    25.12K | 0.015
engine/reference/commandline/service_create.md    |     1 |    60.19K | 0.014
engine/reference/commandline/build.md             |     1 |    36.78K | 0.010
engine/reference/commandline/buildx_build.md      |     1 |    10.24K | 0.009
engine/reference/commandline/stop.md              |     1 |     1.24K | 0.009
engine/reference/commandline/container_start.md   |     1 |     7.09K | 0.008
engine/reference/commandline/builder_build.md     |     1 |    10.05K | 0.008
engine/reference/commandline/image_build.md       |     1 |    11.35K | 0.008
engine/reference/commandline/volume_ls.md         |     1 |     6.53K | 0.007
engine/reference/commandline/plugin_enable.md     |     1 |     4.17K | 0.007
engine/reference/commandline/manifest_annotate.md |     1 |     2.82K | 0.007
samples/index.md                                  |     1 |    14.18K | 0.007
engine/reference/commandline/app.md               |     1 |     4.11K | 0.007
engine/reference/commandline/app_push.md          |     1 |     5.12K | 0.006
engine/reference/commandline/swarm_init.md        |     1 |    13.08K | 0.006
engine/reference/commandline/service_scale.md     |     1 |     5.72K | 0.006
engine/reference/commandline/container_update.md  |     1 |     9.17K | 0.006
engine/reference/commandline/cluster_create.md    |     1 |     3.73K | 0.006
engine/reference/commandline/buildx_create.md     |     1 |     5.17K | 0.006
engine/reference/commandline/container_logs.md    |     1 |     6.96K | 0.006
engine/reference/commandline/buildx_imagetools.md |     1 |     3.61K | 0.006
engine/reference/commandline/app_install.md       |     1 |     6.88K | 0.006
engine/reference/commandline/container_stats.md   |     1 |     6.39K | 0.006
engine/reference/commandline/container_attach.md  |     1 |     6.22K | 0.006
engine/reference/commandline/container_rm.md      |     1 |     6.20K | 0.006
engine/reference/commandline/node_inspect.md      |     1 |     7.64K | 0.006
engine/reference/commandline/stack_ps.md          |     1 |    14.78K | 0.006
engine/reference/commandline/volume_create.md     |     1 |     5.46K | 0.006
engine/reference/commandline/container_prune.md   |     1 |    10.42K | 0.006
engine/reference/commandline/network_create.md    |     1 |    14.63K | 0.006
engine/reference/commandline/app_render.md        |     1 |     5.58K | 0.006

                    done in 46.577 seconds.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -42,78 +65,7 @@ defaults:
       toc_min: 2
       toc_max: 3
 
-  # Set the correct edit-URL for upstream resources. We usually don't create a direct
-  # edit link for these, and instead point to the directory that contains the file.
-  - scope:
-      path: engine/api/v1.*
```
Comment (thaJeztah) [inline _config.yml:48]: This one turned out to be a big bottleneck 😞

**Comment by thaJeztah** [comment]
@usha-mandya you'll like this one 😅

**Comment by usha-mandya** [comment]
@thaJeztah This is super exciting!!

**Review by usha-mandya — APPROVED** [review]

**Comment by thaJeztah** [comment]
Argh! I missed adding one commit for the landing page; double checking now, and will do a quick follow-up

**Comment by thaJeztah** [comment]
Fix in https://github.com/docker/docker.github.io/pulls

**Pull request merged**